apis/nucleus/src/components/__tests__/nebulaapp.inspect.jsx

```diff
@@ -1,119 +1,138 @@
 /* eslint-disable react/jsx-no-constructed-context-values */
+/* eslint-disable no-import-assign */
 import React from 'react';
 import { create, act } from 'react-test-renderer';
+import * as reactDomModule from 'react-dom';
 import { ThemeProvider, StyledEngineProvider, createTheme } from '@nebula.js/ui/theme';
+import * as InstanceContextModule from '../../contexts/InstanceContext';
+import * as useAppSelectionsModule from '../../hooks/useAppSelections';
 
-const mockedReactDOM = { render: sinon.spy() };
-const [{ default: boot, NebulaApp }] = aw.mock(
-  [
-    [require.resolve('react-dom'), () => mockedReactDOM],
-    [require.resolve('../../hooks/useAppSelections'), () => () => [{}]],
-  ],
-  ['../NebulaApp']
-);
+import boot, { NebulaApp } from '../NebulaApp';
 
-const InstanceContext = React.createContext();
+jest.mock('react-dom', () => ({
+  render: jest.fn(),
+}));
 
 describe('Boot NebulaApp', () => {
-  let sandbox;
+  let InstanceContext;
+  let mockedReactDOM;
+  let renderMock;
 
   beforeEach(() => {
-    mockedReactDOM.render.resetHistory();
-    sandbox = sinon.createSandbox();
+    renderMock = jest.fn();
+
+    InstanceContext = React.createContext();
+    InstanceContextModule.default = InstanceContext;
+    jest.spyOn(useAppSelectionsModule, 'default').mockImplementation(() => [{}]);
+    jest.spyOn(reactDomModule, 'render').mockImplementation(renderMock);
+
     const mockedElement = {
       style: {},
-      setAttribute: sandbox.spy(),
-      appendChild: sandbox.spy(),
+      setAttribute: jest.fn(),
+      appendChild: jest.fn(),
     };
     global.document = {
-      createElement: sandbox.stub().returns(mockedElement),
+      createElement: jest.fn().mockReturnValue(mockedElement),
       body: mockedElement,
     };
   });
   afterEach(() => {
-    sandbox.restore();
     delete global.document;
+    jest.restoreAllMocks();
+    jest.resetAllMocks();
   });
 
-  it('should call ReactDOM render', () => {
+  test('should call ReactDOM render', () => {
     boot({ app: { id: 'foo' } });
-    expect(mockedReactDOM.render.callCount).to.equal(1);
+    expect(renderMock).toHaveBeenCalledTimes(1);
   });
-  it('should return api', () => {
+  test('should return api', () => {
     const [api] = boot({ app: { id: 'foo' } });
-    expect(api.add).to.be.a('function');
-    expect(api.remove).to.be.a('function');
-    expect(api.setMuiThemeName).to.be.a('function');
-    expect(api.context).to.be.a('function');
+    expect(api.add instanceof Function).toBe(true);
+    expect(api.remove instanceof Function).toBe(true);
+    expect(api.setMuiThemeName instanceof Function).toBe(true);
+    expect(api.context instanceof Function).toBe(true);
   });
-  it('should add component', async () => {
+
+  test.skip('should add component', async () => {
```

Comment:
for this test and next 5 ones also , literally i could not fine a way to implement the:
```
await act(() => {
      mockedReactDOM.render.callArg(2);
      api.add('foo');
      return rendered;
});
```
in jest, specially the call of `mockedReactDOM.render.callArg(2);`. this seems to be mystery to implement. 

I debugged the code for over 3 hours and figured out that in afterwork seems like the test [passes this line](https://github.com/qlik-oss/nebula.js/blob/afebec0d26b721038278fbcf7a877b3c5960582a/apis/nucleus/src/components/NebulaApp.jsx#L80) and test pass successfully. The `renderer` variable is a promises that we are awaiting it before invoking `appRef.current.addComponent(component);` But in jest test, if we skip this `mockedReactDOM.render.callArg(2);` (because there is no way that i aware of that we could implement that kind of behaviour in jest), we will never reach the `.addComponent(...)` call in our test. 

the following 5 tests also does kinda same thing with other methods, which i marked them to being skipped by jest until we come up with a solution.

Reply:
I think (but have not tested) that
```
mockedReactDOM.render.callArg(2);
```
should be
```
mockedReactDOM.render.mockImplementation((element, container, callback) => {
   callback();
});
```
in jest (maybe also only call callback if not undefined)

Reply:
Yeeeessss, absolutely it is. 
Definitely I learned some thing new today and i would take it for granted. A **big thanks** for this @T-Wizard 🙏 💯